js/filter.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+window.filter = (function () {
+  var filterContainer = document.querySelector('.map__filters-container');
+  var filterTypeElement = filterContainer.querySelector('#housing-type');
+  var filterPriceElement = filterContainer.querySelector('#housing-price');
+  var filterRoomsElement = filterContainer.querySelector('#housing-rooms');
+  var filterGuestsElement = filterContainer.querySelector('#housing-guests');
+  var housingFeaturesCheckboxes = document.querySelector('#housing-features').querySelectorAll('.map__checkbox');
+
+  function filterAds(unfilteredData) {
+    var type = filterTypeElement.value;
+    var price = filterPriceElement.value;
+    var rooms = filterRoomsElement.value;
+    var guests = filterGuestsElement.value;
+    var featuresArray = [];
+
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        featuresArray.push(elem.value);
+      }
+    });
+
+    var result = unfilteredData.slice();
+
+    // Property Filter
+    if (type !== 'any') {
+      result = result.filter(function (ad) {
+        return ad.offer.type === type;
+      });
+    }
+
+    // Price Filter
+    if (price !== 'any') {
+      result = result.filter(function (ad) {
+        var returnProperty = {
+          'middle': function () {
+            return (ad.offer.price >= 10000 && ad.offer.price <= 50000);
+          },
+          'low': function () {
+            return ad.offer.price < 10000;
+          },
+          'high': function () {
+            return ad.offer.price > 50000;
+          }
+        };
+        return returnProperty[price]();
+      });
+    }
+
+    // Rooms Filter
+    if (rooms !== 'any') {
+      result = result.filter(function (ad) {
+        return ad.offer.rooms === +rooms;
+      });
+    }
+
+    // Guests Filter
+    if (guests !== 'any') {
+      result = result.filter(function (ad) {
+        if (+guests === 0 && ad.offer.guests > 2) {
+          return ad;
+        }
+        return ad.offer.guests === +guests;
+      });
+    }
+
+    // Features Filter
+    if (featuresArray.length <= 0) {
+      result = result.slice();
+    } else {
+      result = result.filter(function (ad) {
```

Comment:
you missed a very important comment, my bad I didn't leave in the code
![image](https://user-images.githubusercontent.com/1570578/52462568-8129b500-2b6b-11e9-9a80-10060700c455.png)

Reply:
Let's do filtering with just one filter function.